ivanti_security_controls/icon_ivanti_security_controls/actions/get_scanned_machine_details/action.py

```diff
@@ -0,0 +1,37 @@
+import insightconnect_plugin_runtime
+from .schema import GetScannedMachineDetailsInput, GetScannedMachineDetailsOutput, Input, Output, Component
+
+# Custom imports below
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+
+class GetScannedMachineDetails(insightconnect_plugin_runtime.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='get_scanned_machine_details',
+                description=Component.DESCRIPTION,
+                input=GetScannedMachineDetailsInput(),
+                output=GetScannedMachineDetailsOutput())
+
+    def run(self, params={}):
+        machine_id = ""
+        scan_id = params.get(Input.SCAN_ID)
+        hostname = params.get(Input.HOSTNAME)
+
+        patch_scan_machine = self.connection.ivanti_api.get_patch_scan_machines(scan_id)
+
+        for machine in patch_scan_machine['value']:
```

Comment:
Since this action is to get the results of a single machine (by hostname) of a scan, could we return the machine details at the top level instead of nested?

For example, instead of:
```
{
"detected_patches": [],
"patch_scan_machine": {
    "count": 1,
    "links": {
      "next": {
        "href": "https://ivanti-w16.vuln.lax.rapid7.com:3121/st/console/api/v1.0/patch/scans/f447bd51-de32-4bd6-a28e-ad834694d5ac/machines?count=10\u0026start=73"
      }
    },
    "value": [
      {
        "completedOn": "2020-05-12T21:53:57.71Z",
        "domain": "WORKGROUP",
        "errorNumber": 0,
        "id": 72,
        "installedPatchCount": 16,
        "links": {
          "patches": {
            "href": "https://ivanti-w16.vuln.lax.rapid7.com:3121/st/console/api/v1.0/patch/scans/f447bd51-de32-4bd6-a28e-ad834694d5ac/machines/72/patches"
          }
        },
        "missingPatchCount": 3,
        "missingServicePackCount": 1,
        "name": "ivanti-w16"
      }
    ]
  }
}
```

We would return:
```
{
       "detected_patches": [],
        "completedOn": "2020-05-12T21:53:57.71Z",
        "domain": "WORKGROUP",
        "errorNumber": 0,
        "id": 72,
        "installedPatchCount": 16,
        "links": {
          "patches": {
            "href": "https://ivanti-w16.vuln.lax.rapid7.com:3121/st/console/api/v1.0/patch/scans/f447bd51-de32-4bd6-a28e-ad834694d5ac/machines/72/patches"
          }
        },
        "missingPatchCount": 3,
        "missingServicePackCount": 1,
        "name": "ivanti-w16"
      }
```

Reply:
That makes sense. I will update the action.